client/src/util/stateManager/colorHelpers.ts

```diff
@@ -1,41 +1,47 @@
-/*
-Helper functions for the embedded graph colors
-*/
+/**
+ * Helper functions for the embedded graph colors
+ */
 import * as d3 from "d3";
 import { interpolateRainbow, interpolateCool } from "d3-scale-chromatic";
 import memoize from "memoize-one";
+import { Action } from "redux";
 import * as globals from "../../globals";
 import parseRGB from "../parseRGB";
 import { range } from "../range";
-import { Dataframe, LabelType } from "../dataframe";
+import { Dataframe, DataframeValueArray, LabelType } from "../dataframe";
+import { Field, Schema } from "../../common/types/schema";
+import { Query } from "../../annoMatrix/query";
+import { Genesets } from "../../reducers/genesets";
 
-/*
-given a color mode & accessor, generate an annoMatrix query that will
-fulfill it
-*/
-// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types --- FIXME: disabled temporarily on migrate to TS.
+interface Colors {
+  // cell label to color mapping
+  rgb: [number, number, number][];
+  // function, mapping label index to color scale
+  scale: d3.ScaleSequential<string, never> | undefined;
+}
+
+/**
+ * Given a color mode & accessor, generate an annoMatrix query that will
+ * fulfill it
+ */
 export function createColorQuery(
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  colorMode: any,
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  colorByAccessor: any,
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  schema: any,
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  genesets: any
-) {
+  colorMode: Action["type"],
```

Comment:
Why `Action["type"]`? Isn't this just `any`

Reply:
Technically `Action["type"]` is generic and can be strings or symbols. I opted to use `Action["type"]` here instead of `string` because I wanted to signal that `colorMode` comes from action type directly, which `string` can't convey such relationship. What do you think?

Reply:
Since colorMode is passed from reducer state and not the action, shouldn't it be typed as ColorState["colorMode"] ?

Reply:
Ooh great point! Added `ColorsState` 👌 Thank you!